src/rust/engine/logging/src/executor.rs

```diff
@@ -0,0 +1,130 @@
+use futures::Future;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+
+// TODO: It's strange that this is an exposed interface from the logging crate, rather than an
```

Comment:
Is the only reason it is in the crate so that it can have access to `set_destination`? It feels like it would be ok to make that method public with a sufficiently large docstring. But not a blocker.

Reply:
`set_destination` is already public - the alternative would be to move it to a crate of its own - would you prefer that?

Reply:
I think so, yea. But not a blocker.